controllers/topics/index.js

```diff
@@ -84,13 +85,32 @@ const putFollowTopic = async (req, res) => {
 
 const getTopics = async (req, res) => {
   const {name} = req.query;
+  const signUserId = req.userId;
+  const anonymousId = await getAnonymUser(req.userId);
 
   try {
     const results = await sequelize.query(
       `SELECT 
             "Topic".*,
             count("topicFollower"."user_id") 
-                AS "followersCount"
+                AS "followersCount",
+            CASE
+              WHEN (SELECT 
+                count(user_id) as user_id_total 
+              FROM user_topics 
+              WHERE 
+                user_id = :signUserId
+                AND topic_id = "Topic".topic_id 
+              LIMIT 1) > 0 THEN 'sign'
+              WHEN (SELECT 
+                count(user_id) as user_id_total 
+              FROM user_topics 
+              WHERE 
+                user_id = :anonymousId
+                AND topic_id = "Topic".topic_id 
+              LIMIT 1) > 0 THEN 'anon'
+              ELSE 'null'
+              END AS is_followed_by
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query has been extended to include a new column `is_followed_by`, which indicates whether the topic is followed by a signed-in user or an anonymous user. The logic seems correct, but there are two subqueries that could potentially slow down the performance if there are many topics and users. Consider optimizing this part by reducing the number of subqueries or using JOINs instead.



<!-- This is an auto-generated comment by OSS CodeRabbit -->